pkg/mimirtool/commands/rules_test.go

```diff
@@ -115,6 +117,105 @@ func TestCheckDuplicates(t *testing.T) {
 	}
 }
 
+func TestRuleCommand_checkRules(t *testing.T) {
+	completelyBadRuleName := rulefmt.RuleNode{
+		Record: yaml.Node{Value: "up", Kind: yaml.ScalarNode},
+		Expr:   yaml.Node{Value: "up==1", Kind: yaml.ScalarNode},
+	}
+	strictlyBadRuleName := rulefmt.RuleNode{
+		Record: yaml.Node{Value: "up:onecolonmissing", Kind: yaml.ScalarNode},
+		Expr:   yaml.Node{Value: "up==1", Kind: yaml.ScalarNode},
+	}
+	validRule1 := rulefmt.RuleNode{
+		Record: yaml.Node{Value: "rule:up:nothing", Kind: yaml.ScalarNode},
+		Expr:   yaml.Node{Value: "up==1", Kind: yaml.ScalarNode},
+	}
+	validRule2 := rulefmt.RuleNode{
+		Record: yaml.Node{Value: "rule:down:nothing", Kind: yaml.ScalarNode},
+		Expr:   yaml.Node{Value: "up==0", Kind: yaml.ScalarNode},
+	}
+	for _, tc := range []struct {
+		name       string
+		rules      []rulefmt.RuleNode
+		strict     bool
+		shouldFail bool
+	}{
+		{
+			name:       "completely rule name, not strict fals too",
+			rules:      []rulefmt.RuleNode{validRule1, completelyBadRuleName, validRule2},
+			strict:     false,
+			shouldFail: true,
+		},
+		{
+			name:       "bad rule name, not strict",
+			rules:      []rulefmt.RuleNode{validRule1, strictlyBadRuleName, validRule2},
+			strict:     false,
+			shouldFail: false,
+		},
+		{
+			name:       "bad rule name, strict",
+			rules:      []rulefmt.RuleNode{validRule1, strictlyBadRuleName, validRule2},
+			strict:     true,
+			shouldFail: true,
+		},
+		{
+			name:       "strictly rule name, not strict",
+			rules:      []rulefmt.RuleNode{validRule1, strictlyBadRuleName, validRule2},
+			strict:     false,
+			shouldFail: false,
+		},
+		{
+			name:       "no duplicates, strict",
+			rules:      []rulefmt.RuleNode{validRule1, validRule2},
+			strict:     true,
+			shouldFail: false,
+		},
+		{
+			name:       "with duplicates, not strict",
+			rules:      []rulefmt.RuleNode{validRule1, validRule1},
+			strict:     false,
+			shouldFail: false,
+		},
+		{
+			name:       "with duplicates, strict",
+			rules:      []rulefmt.RuleNode{validRule1, validRule1},
+			strict:     true,
+			shouldFail: true,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			file := filepath.Join(t.TempDir(), "rules.yaml")
+			{
+				// Setup.
+				f, err := os.Create(file)
+				require.NoError(t, err)
+				contents := rules.RuleNamespace{
+					Namespace: "test",
+					Groups: []rwrulefmt.RuleGroup{{
+						RuleGroup: rulefmt.RuleGroup{
+							Name:  "rulegroup",
+							Rules: tc.rules,
+						},
+					}},
+				}
+				require.NoError(t, yaml.NewEncoder(f).Encode(contents))
+				require.NoError(t, f.Close())
```

Comment:
I suggest you use `t.Cleanup` instead. I don't think the test should fail either if closing the file fails for whatever reason.
```suggestion
```

Reply:
I want to make sure that file is closed before we read it, so leaving this here.

Reply:
Makes sense.